### PR TITLE
feat: added customAnnotations to RBAC Manager

### LINKED ISSUE #419: Allow to add custom annotations to RBAC Manager.
### What problem are you facing?

I need to add custom annotations to RBAC Manager. 
There is little more to add.

### How could UXP help solve your problem?

I am directly preparing a PR to resolve this.

### PR DESCRIPTION
**Pull request by pierluigilenoci**
### Description of your changes

This change allows to add custom annotations to RBAC Manager.

Fixes #419

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@turkenf I'm sorry, but I can't add the proper label. 😢

File: cluster/charts/universal-crossplane/templates/crossplane/rbac-manager-deployment.yaml [hunk 1]
```diff
@@ -8,6 +8,9 @@ metadata:
     app: {{ template "crossplane.name" . }}-rbac-manager
     release: {{ .Release.Name }}
     {{- include "crossplane.labels" . | indent 4 }}
+  {{- with .Values.customAnnotations }}
```
Comment (turkenh) [inline cluster/charts/universal-crossplane/templates/crossplane/rbac-manager-deployment.yaml:11]: @pierluigilenoci thanks for the PR. 

We are inheriting the helm chart manifests from upstream, so any changes we made here would be overridden with `make generate`. Would you mind opening against crossplane/crossplane changing the following file? https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/rbac-manager-deployment.yaml#L10

**Comment by pierluigilenoci** [comment]
@turkenh like this https://github.com/crossplane/crossplane/pull/5243 ?

**Comment by turkenh** [comment]
> @turkenh like this [crossplane/crossplane#5243](https://github.com/crossplane/crossplane/pull/5243) ?

Yes. Thanks @pierluigilenoci! Closing this one in favor of the other. 
Changes will be reflected here in the upcoming release.

**Pull request closed without merging**